### DISCUSSION TITLE
GitHub-hosted runners VM size

### DISCUSSION THREAD

**Discussion by teqdruid** [community/community#26678] [Actions]
<p>Hi All-</p>
<p>The GitHub-host runners only support a woefully underpowered VM type. (2 cores, 7GB RAM.) I know that self-hosted runners are an option, but they’re not recommended for public projects. For large projects (like <a href="https://github.community/t/bigger-github-hosted-runners-disk-space/17267/7">LLVM</a>) or ones that have to build large projects, this isn’t ideal.</p>
<p>Is there a plan to address this need? This could be by providing a secure option for self-hosted runner VMs on public repos (with shutdown and scaling support to save $$). Could also be by paying for beefier machines. Obviously, dealing with PRs from other repos would have to be controlled.</p>
<p>~John</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/teqdruid">@teqdruid</a>,</p>
<p>Thanks for your feedback.</p>
<p>I recommend that you can directly share your suggestions <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="noopener nofollow ugc">here</a>. That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>
<p>In addition, you also can follow the “<a href="https://github.com/github/roadmap" rel="noopener nofollow ugc"><strong>GitHub public roadmap</strong></a>” repository where you can learn about what new features the appropriate engineering teams are working on, what stage these features are in, and when the teams expect to bring them to users.</p>

**Comment by byrneh** [comment, marked as answer]
<p>Hi <a class="mention" href="/teqdruid">@teqdruid</a>, I think this is a well known use case, I suspect this future roadmap feature below may deliver what you need</p>
<aside class="onebox githubissue">
  <header class="source">
      <a href="https://github.com/github/roadmap/issues/95" target="_blank" rel="noopener nofollow ugc">github.com/github/roadmap</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/github/roadmap/issues/95" target="_blank" rel="noopener nofollow ugc">Actions: Multiple hosted runner sizes (Beta)</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2020-07-24" data-time="17:16:38" data-timezone="UTC">05:16PM - 24 Jul 20 UTC</span>
      </div>


      <div class="user">
        <a href="https://github.com/github-product-roadmap" target="_blank" rel="noopener nofollow ugc">
          <img alt="github-product-roadmap" src="https://user-images.githubusercontent.com/39440862/181099162-f1504be7-cabd-4b7c-9b03-2a697b982dec.png" class="onebox-avatar-inline" width="20" height="20">
          github-product-roadmap
        </a>
      </div>
    </div>
  </div>
</div>

<div class="github-row">
  <p class="github-content">Summary 
GitHub hosted runners will come in new more powerful configurations which can include more CPU cores, RAM, disk space, and...</p>
</div>

<div class="labels">
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">actions</span>
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">all</span>
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">beta</span>
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">cloud</span>
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">code-to-cloud</span>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by teqdruid** [comment]
<p>Excellent! That’s exactly what I asking about!</p>
<p>Would this be available to public repos? How would pricing work for them?</p>

**Comment by byrneh** [comment]
<p><a class="mention" href="/teqdruid">@teqdruid</a>,  unfortunately the public roadmap item has moved moved to Future column so that is 3Q 2021 or later, (unless it gets moved earlier), so I would not expect much additional detail for a good while yet.</p>
<p>I guess that leaves you with the answer of self-hosted runners, which are not recommended with public repositories for reasons already stated by GitHub, and which you have already noted.<br>
However this is not an issue with GitHub-hosted runners because each GitHub-hosted runner is always a clean isolated virtual machine, and it is destroyed at the end of the job execution.<br>
So if you were able to implement some self-hosted runners with same operating model (always clean and isolated and destroyed at end of jo execution) as GitHubs own default runners do your risk position should be unchanged.<br>
I appreciate this may be easier said than done (and costly), I am not an Actions user yet so cannot comment on technicalities of the suggestion.</p>

**Comment by Jongy** [comment]
<p>Hi, did this get resolved? Can I have my GitHub hosted runners with a different machine size? 2 CPUs 7GB are not enough :frowning:</p>

**Comment by byrneh** [comment]
<p><a class="mention" href="/jongy">@Jongy</a> unfortunately current sizings are still as listed here <a href="https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources" rel="noopener nofollow ugc">supported-runners-and-hardware-resources</a></p>
<p>I can’t see anything on the Public Roadmap for this, so its a matter of watching the roadmap and announcements and using self-hosted runners sized to your needs unless someone knows differently. I wouldn’t assume if will be a free offering if/when something is delivered.</p>
<hr>
<p>In this repository, you can find the <a href="https://github.com/github/feedback/discussions" rel="noopener nofollow ugc">official GitHub public feedback discussions</a>, you could add your voice hear.<br>
An example of similar requirement you could add your voice to is here<br>
<a href="https://github.com/github/feedback/discussions/8358" rel="noopener nofollow ugc">Please let us pay for a faster CI box for Github Workflows #8358</a></p>

**Comment by danielcompton** [comment]
<p>It looks like this is on the roadmap under:</p>
<aside class="onebox githubissue" data-onebox-src="https://github.com/github/roadmap/issues/161">
  <header class="source">

      <a href="https://github.com/github/roadmap/issues/161" target="_blank" rel="noopener nofollow ugc">github.com/github/roadmap</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/github/roadmap/issues/161" target="_blank" rel="noopener nofollow ugc">Actions: More powerful runners choices for GitHub-managed runners on GitHub.com</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2021-02-10" data-time="16:02:51" data-timezone="UTC">04:02PM - 10 Feb 21 UTC</span>
      </div>


      <div class="user">
        <a href="https://github.com/github-product-roadmap" target="_blank" rel="noopener nofollow ugc">
          <img alt="github-product-roadmap" src="https://user-images.githubusercontent.com/811954/181099169-5714fde9-1c74-4e4d-96db-ca900662764c.png" class="onebox-avatar-inline" width="20" height="20">
          github-product-roadmap
        </a>
      </div>
    </div>

    <div class="labels">
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          actions
        </span>
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          cloud
        </span>
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          code-to-cloud
        </span>
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          github enterprise
        </span>
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          beta
        </span>
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          github team
        </span>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">**Summary** 
GitHub-managed runners will now include the option to choose more <span class="show-more-container"><a href="" rel="noopener" class="show-more">…</a></span><span class="excerpt hidden">powerful combinations of CPU, RAM, &amp; disk space for your jobs. These more powerful configurations are also privately pooled on your behalf, which means you can optionally enable a static Public IP range for use with your firewall, or proxy/bastion server.

**Intended Outcome**
Our current runners are static in configuration and can be limiting for more intensive or specialized workflows. These new configurations will make running more complicated workflows easier and faster.

**How will it work?** 
You will create as many different runner configurations as you need. They will have custom names, labels, runner groups, and hardware configurations. More powerful runners will come with higher prices per minute but they are fully managed by GitHub.

![image](https://user-images.githubusercontent.com/18743761/131022241-995d324d-c1d4-415b-be12-205dcee9cdc4.png)</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Reply by FezVrasta** [reply]
Is there any update on this? Any way to opt into the beta somehow? I would need it for a private repository.

**Reply by kennysong** [reply]
Same question.

**Comment by stiv-yakovenko** [comment]
Any progress?
